### ISSUE TITLE
Data source wrapper

### ISSUE THREAD

**Issue by yoavnash** [simphony/osp-core#620]
So far we had wrappers for simulations and for data sink (to store data). We still don't have a wrapper for extracting data from an existing data repo. Such a wrapper would be beneficial for demos and various applications.

Suggestions:
- [NOMAD](https://nomad-lab.eu/services/repo-arch) Wrapper
- [OPTIMADE](https://www.optimade.org/) wrapper

**Comment by pablo-de-andres** [comment]
Do we just want to extract data? Because then I would not use a wrapper, since both simulations and data storage can be used for communication in both directions which I think is part of what makes wrappers wrappers

**Comment by yoavnash** [comment]
Good point. It can still be beneficial though to have a "read-only" wrapper, for direct access to data in an OSP-core script.

**Comment by pablo-de-andres** [comment]
I think would define another component at the same level as wrappers. A parser/reader, if you will.
And we could divide between data repos, excel files and other formats...

**Comment by yoavnash** [comment]
That sounds indeed better. Great suggestion.

**Comment by pablo-de-andres** [comment]
It might make for a nice symmetric design if we think of _readers_, _writers_ and _wrappers_

**Comment by yoavnash** [comment]
Could be related to #267 and #290

**Comment by yoavnash** [comment]
Could be related to mapping data from tables to RDF. Related technologies:
- OntoRefine: https://openrefine.org/ , https://graphdb.ontotext.com/documentation/free/loading-data-using-ontorefine.html
- YARRRML
- SPARQL-Integrate: https://github.com/SmartDataAnalytics/RdfProcessingToolkit/blob/master/README-SI.md

**Comment by kysrpex** [comment]
I am labelling this also as breaking change as we might want to introduce a new API or do changes to the wrapper API for this.

**Comment by kysrpex** [comment]
Related #651.
